src/libraries/Pool.sol

```diff
@@ -365,7 +369,8 @@ library Pool {
                 ) ? params.sqrtPriceLimitX96 : step.sqrtPriceNextX96,
                 state.liquidity,
                 state.amountSpecifiedRemaining,
-                swapFee
+                // use the effective fee
+                uint24(cache.protocolFee + swapFee - (uint32(cache.protocolFee) * uint32(swapFee)) / MAX_EFFECTIVE_FEE)
```

Comment:
this whole sum can be unchecked - that will save a good amount of gas. Currently this PR adds 1000 gas to a simple swap 😢